logic/resolvable/Concludable.java

```diff
@@ -373,15 +503,41 @@ public Value(final ValueConstraint<?> constraint) {
         }
 
         @Override
-        Stream<Unifier> unify(Rule.Conclusion.Value unifyWith) {
-            Optional<Unifier> unifier = tryExtendUnifier(constraint().owner(), unifyWith.constraint().owner(), new HashMap<>());
-            if (!unifier.isPresent()) return Stream.empty();
-            if (constraint().isVariable() && constraint().asVariable().value().reference().isName()) {
-                unifier = tryExtendUnifier(constraint().asVariable().value(),
-                                           unifyWith.constraint().asVariable().value(), unifier.get().mapping());
-                if (!unifier.isPresent()) return Stream.empty();
+        ResourceIterator<Unifier> unify(Rule.Conclusion.Value valueConclusion, ConceptManager conceptMgr) {
+            Unifier.Builder unifierBuilder = Unifier.builder();
+            if (unificationSatisfiable(constraint().owner(), valueConclusion.constraint().owner())) {
+                unifierBuilder.add(constraint().owner().identifier(), valueConclusion.constraint().owner().identifier());
+            } else return Iterators.empty();
+
+            /*
+            Interesting case:
+            Unifying 'match $x >= $y' with a rule inferring a value: 'then { $x has age 10; }'
+            Conceptually, $x will be 'age 10'. However, for a rule to return a valid answer for this that isn't found
+            via a traversal, we require that $y also be 'age 10'. The correct unification is therefore to map both
+            $x and $y to the same new inferred attribute.
+            Trying to find results for a query 'match $x > $y' will never find any answers, as a rule can only infer
+            an equality, ie. 'then { $x has $_age; $_age = 10; $_age isa age; }
+             */
+            if (constraint().isVariable()) {
+                assert constraint().value() instanceof ThingVariable;
+                ThingVariable value = (ThingVariable) constraint().value();
+                if (unificationSatisfiable(constraint().predicate(), valueConclusion.constraint().predicate())) {
+                    unifierBuilder.add(value.identifier(), valueConclusion.constraint().owner().identifier());
+                } else return Iterators.empty();
+                // } else {
+                // form: $x > 10 -> require $x to satisfy predicate > 10
+                // TODO after restructuring concludables, we whould revisit requirements on `Value` concludables
```

Comment:
this is commented with a note because the immediate follow up to be implemented is to rearchitect reasoner data structures to get rid of this `else`, among with some other inconsistencies/code smells